test/wizard/wizard.spec.js

```diff
@@ -0,0 +1,209 @@
+    element = compileHtml(modalHtml, $scope);
+    $scope.$digest();
+
+    // there are two dependent timeouts in the wizard that need to be flushed
+    $timeout.flush();
+    $timeout.flush();
+  });
+
+  it('should dispatch the cancel event on the close button click', function () {
+    var closeButton = element.find('.close');
+    $rootScope.$on('wizard.done', function (event, data) {
+      expect(data).toBe('cancel');
+    });
+    spyOn($rootScope, '$emit');
+    eventFire(closeButton[0], 'click');
+    $scope.$digest();
+
+    expect($rootScope.$emit).toHaveBeenCalled();
+  });
+
+  it('should have three step indicators in the header', function () {
+    var stepsIndicator = element.find('.wizard-pf-steps .wizard-pf-step');
+    expect(stepsIndicator.length).toBe(3);
+  });
+
+  it('should have two sections in the left-hand pane', function () {
+    var stepsIndicator = element.find('.wizard-pf-sidebar .list-group-item');
+    var hiddenStepsIndicator = element.find('section.ng-hide .wizard-pf-sidebar .list-group-item');
+
+    //find all hidden steps and remove them from total step count to make sure correct number are visible
+    expect(stepsIndicator.length - hiddenStepsIndicator.length).toBe(2);
+  });
+
+  it('should have disabled the next button', function () {
+    var checkDisabled = element.find('.wizard-pf-next').attr('disabled');
+    expect(checkDisabled).toBe('disabled');
+  });
+
+  it('should have enabled the next button after input and allowed navigation', function () {
+    var nextButton = element.find('.wizard-pf-next');
+    var nameBox = element.find('#new-name');
+    nameBox.val('test').triggerHandler('input');
+    eventFire(nextButton[0], 'click');
+    var stepIndicator = element.find('.wizard-pf-sidebar .list-group-item.active .wizard-pf-substep-number');
+    expect(stepIndicator.text()).toBe('1B.');
+  });
+
+  it('should have allowed moving back to first page after input and allowed navigation', function () {
+    var nextButton = element.find('.wizard-pf-next');
+    var nameBox = element.find('#new-name');
+    nameBox.val('test').triggerHandler('input');
+    eventFire(nextButton[0], 'click');
+    var stepIndicator = element.find('.wizard-pf-sidebar .list-group-item.active .wizard-pf-substep-number');
+    expect(stepIndicator.text()).toBe('1B.');
+
+    var backButton = element.find('#backButton');
+    eventFire(backButton[0], 'click');
+    var stepIndicator = element.find('.wizard-pf-sidebar .list-group-item.active .wizard-pf-substep-number');
+    expect(stepIndicator.text()).toBe('1A.');
+  });
+
+  it('should have allowed navigation to review page', function () {
+    var nextButton = element.find('.wizard-pf-next');
+    var nameBox = element.find('#new-name');
+    nameBox.val('test').triggerHandler('input');
+    $scope.$digest();
+
+    $scope.currentStep = 'Review';
+    $scope.$digest();
+    $timeout.flush();
+    $timeout.flush();
```

Comment:
Yes - there are several layers that need to be rendered  and timeouts that cascade - it doesn't show to the naked eye when rendering but they need to be flushed when in a unit test.